File: src/dotnet/ArdsTools/Ards.NetCore.Entity/Models/AntennaPath.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ards.NetCore.Entity.Models
{
    [Table("antenna_path", Schema = "adif")]
    public partial class AntennaPath
    {
        [Column("id")]
        public int Id { get; set; }
        [Required]
        [Column("abbreviation")]
        [StringLength(1)]
        public string Abbreviation { get; set; }
        [Required]
        [Column("meaning")]
        [StringLength(15)]
        public string Meaning { get; set; }
    }
}

File: src/java/ards-adif-api/src/main/java/com/ards/adif/api/model/AntennaPath.java
package com.ards.adif.api.model;

import java.io.Serializable;
import javax.persistence.*;


/**
 * The persistent class for the antenna_path database table.
 * 
 */
@Entity
@Table(name="ANTENNA_PATH")
@NamedQuery(name="AntennaPath.findAll", query="SELECT a FROM AntennaPath a")
public class AntennaPath implements Serializable {
	
	private static final long serialVersionUID = 1L;

	@Id
	@Column(unique=true, nullable=false)
	private Integer id;

	@Column(nullable=false, length=1)
	private String abbreviation;

	@Column(nullable=false, length=15)
	private String meaning;

	public AntennaPath() {
	}

	public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getAbbreviation() {
		return this.abbreviation;
	}

	public void setAbbreviation(String abbreviation) {
		this.abbreviation = abbreviation;
	}

	public String getMeaning() {
		return this.meaning;
	}

	public void setMeaning(String meaning) {
		this.meaning = meaning;
	}

}